src/debug/debug.cc

```diff
@@ -761,8 +762,10 @@ DebugFile::reset(FILE *newFile, const char *newName)
     }
     file_ = newFile; // may be nil
 
-    if (file_)
+    if (file_) {
+        commSetCloseOnExec(fileno(file_));
```

Comment:
The primary reason I marked this PR as a draft is to ask the following question: Do we care about platforms where commSetCloseOnExec() does not work (e.g., because `FD_CLOEXEC` is not defined)?

AFAICT, if we do want to fully support those platforms, then simply calling commSetCloseOnExec() is not enough (or, arguably, should _never_ be needed!) because we have to loop through all open descriptors and close all of them explicitly before calling one of the exec*() functions anyway.

If we do not care, then it would be nice to add a comment to commSetCloseOnExec(), saying as much.

Reply:
@yadij, @kinkie, any opinion regarding commSetCloseOnExec()? If there are none, then I will proceed with the assumption that "we do not care" but will not "add a comment to commSetCloseOnExec()" since I got no explicit confirmation that we do not actually care, and I am not sure whether we should.

Reply:
according to https://www.gnu.org/software/gnulib/manual/html_node/fcntl_002eh.html , FD_CLOEXEC is undefined on MSVC 14 (an incredibly old version) and mingw. The former is incredibly old, the latter might be relevant for us at some point in the future but I for now I would just state that we could not care for platforms where it's not defined.

Reply:
@yadij, do you agree with @kinkie suggestion to state that we do not care for platforms where FD_CLOEXEC is not defined? This draft PR is blocked on making that decision as detailed in this change request.

Reply:
In addition to my previous statement, according to https://devblogs.microsoft.com/oldnewthing/20111216-00/?p=8873 , FD_CLOEXEC is the default behaviour on Windows. 

In other words, let's move forward and focus on the platforms we have, the discussion about how to better deal with windows belongs there

Reply:
> let's move forward and focus on the platforms we have, the discussion about how to better deal with windows belongs there

I have pushed the corresponding documentation change in commit c82d196 and am marking this PR as ready for review.

Reply:
Looks good to me, the comment is clear and sets clear responsibilities. #shipit